File: python/917.py
class Solution:
    def reverseOnlyLetters(self, s: str) -> str:
        letters = ''
        res = ''
        i = len(letters) - 1
        for letter in s:
            if letter.isalpha():
                letters += letter
        print(letters)
        for char in s:
            if not char.isalpha():
                res += char
            else:
                res += letters[i]
                i -= 1
        return res
    
print(Solution().reverseOnlyLetters("a-bC-dEf-ghIj"))

File: javaScript/917.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseOnlyLetters = function(s) {
    let str = ''
    let i = str.length - 1
    let res = ''
    for(let char of s){
        if (char.toLowerCase() != char.toUpperCase()) {
            str += char
        }
    }    
    for(let char of s){
        if (char.toLowerCase() != char.toUpperCase()) {
            res += str[i]
            i--
        } else {
            res += char
        }
    }
    return res
};


console.log(reverseOnlyLetters("a-bC-dEf-ghIj"));
